low-liquidity-market-attack-monitor/src/agent.js

```diff
@@ -0,0 +1,204 @@
+const config = require('../agent-config.json');
+
+// set up a variable to hold initialization data used in the handler
+const initializeData = {};
+
+// the Forta SDK filterLog function does not supply the logIndex with the results
+//  we need this to determine the order the events were emitted in
+/* eslint-disable */
+function filterLog(eventLogs, eventAbi, contractAddress) {
+  var logs = eventLogs;
+  eventAbi = Array.isArray(eventAbi) ? eventAbi : [eventAbi];
+  if (contractAddress) {
+      contractAddress = Array.isArray(contractAddress)
+          ? contractAddress
+          : [contractAddress];
+      var contractAddressMap_1 = {};
+      contractAddress.forEach(function (address) {
+          contractAddressMap_1[address.toLowerCase()] = true;
+      });
+      logs = logs.filter(function (log) { return contractAddressMap_1[log.address.toLowerCase()]; });
+  }
+  var results = [];
+  var iface = new ethers.utils.Interface(eventAbi);
+  for (var _i = 0, logs_1 = logs; _i < logs_1.length; _i++) {
+      var log = logs_1[_i];
+      try {
+          var parsedLog = iface.parseLog(log);
+          results.push(Object.assign(parsedLog, { address: log.address, logIndex: log.logIndex }));
+      }
+      catch (e) { }
+  }
+  return results;
+}
+/* eslint-enable */
+
+// helper function to create cToken alerts
+function createMarketAttackAlert(
+  protocolName,
+  protocolAbbreviation,
+  developerAbbreviation,
+  compTokenSymbol,
+  compTokenAddress,
+  mintAmount,
+  mintTokens,
+  maliciousAddress,
+  maliciousAmount,
+) {
+  const finding = Finding.fromObject({
+    name: `${protocolName} cToken Market Attack Event`,
+    description: `The address ${maliciousAddress} is potentially manipulating the cToken ${compTokenSymbol} market`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-MARKET-ATTACK-EVENT`,
+    type: FindingType.Suspicious,
+    severity: FindingSeverity.Info,
+    protocol: protocolName,
+    metadata: {
+      compTokenSymbol,
+      compTokenAddress,
+      mintAmount,
+      mintTokens,
+      maliciousAddress,
+      maliciousAmount,
+    },
+  });
+  return finding;
+}
+
+async function getCompoundTokens(
+  provider,
+  comptrollerContract,
+  compTokenAbi,
+  excludeAddresses,
+  compTokens,
+) {
+  let compTokenAddresses = await comptrollerContract.getAllMarkets();
+  compTokenAddresses = compTokenAddresses
+    .map((addr) => addr.toLowerCase())
+    .filter((addr) => excludeAddresses.indexOf(addr) === -1)
+    .filter((addr) => !Object.keys(compTokens).includes(addr));
+
+  const promises = compTokenAddresses.map(async (tokenAddress) => {
+    const contract = new ethers.Contract(tokenAddress, compTokenAbi, provider);
+    const symbol = await contract.symbol();
+    const underlying = await contract.underlying();
+
+    // eslint-disable-next-line no-param-reassign
+    compTokens[tokenAddress] = {
+      symbol,
+      underlying,
+    };
+  });
+  await Promise.all(promises);
```

Comment:
``
```suggestion
  await Promise.all(compTokenAddresses.map(async (tokenAddress) => {
    const contract = new ethers.Contract(tokenAddress, compTokenAbi, provider);
    const symbol = await contract.symbol();
    const underlying = await contract.underlying();

    // eslint-disable-next-line no-param-reassign
    compTokens[tokenAddress] = {
      symbol,
      underlying,
    };
  }));
```
If you aren't doing any other async calls, then you can simplify a little bit right here. You should double check with @echoAE if my recommendation is OK or should we stick with our original patterns.